### DISCUSSION TITLE
Problems deploying a trame app using VtkRemoveView in Docker (with cookiecutter)

### DISCUSSION THREAD

**Discussion by loisvazquezind** [Kitware/trame#433] [General] [answered]
Hello.

I have been trying to deploy the finite element analysis example (https://github.com/Kitware/trame/tree/master/examples/06_vtk/Applications/FiniteElementAnalysis) in Docker using the cookiecutter (https://github.com/Kitware/trame-cookiecutter) tool. When I deploy the app using the `VtkLocalView`, all works fine, but when I deploy it using the `VtkRemoveView`, the application does not render well: I almost access the app but I got the "Connection closed" spinner. Is there any limitation for deploying a VtkRemoveView in Docker or I could have another error?

Thanks in advance.

**Comment by jourdain** [comment, marked as answer]
Default VTK can not work within docker. In order to solve your problem you need to either use vtk-osmesa (CPU) or vtk-egl (GPU). 

You can see a simple docker example using vtk [here](https://github.com/Kitware/trame/tree/master/examples/deploy/docker/VtkRendering).

**Reply by evdkmv** [reply]
@loisvazquezind can you run the mentioned example without errors?

**Reply by loisvazquezind** [reply]
Hi. Yes, I can try and it works. I also try to run it with cookie-cutter and also works. Thanks.

**Reply by loisvazquezind** [reply]
Hello again.
I deployed the example on the GPU following the instructions described [here](https://github.com/Kitware/trame/tree/master/examples/deploy/docker/VtkRendering), but the performance in the remote view is even worse than using the CPU I'm using vtk-egl instead of vtk-osmesa and using the 'kitware/trame:glvnd' as the base image in the Dockerfile instead of 'kitware/trame:py3.9'. Do I need to configure anything else?

**Reply by jourdain** [reply]
https://kitware.github.io/trame/guide/deployment/docker.html#egl-caution

**Reply by loisvazquezind** [reply]
Thanks, that works!
